### ISSUE TITLE
MyPy cannot infer the type of an empty `dict` that should be an instance of `TypedDict`

### ISSUE THREAD

**Issue by andrewsonin** [python/mypy#10468]
**Bug Report**

MyPy cannot figure out the type of an empty nested `dict` (which should be typed) in one case, but correctly infers it in a very similar case.

**To Reproduce**

```python
from collections import deque
from typing import TypedDict, Dict, Deque


class MyTypedDict(TypedDict):
    pass


Step = Dict[int, MyTypedDict]


a: Deque[Step] = deque([{}])   # OK
b: Deque[Step] = deque(({},))  # MyPy error
```

MyPy outputs the following for the latter `deque`:
`error: Argument 1 to "deque" has incompatible type "Tuple[Dict[<nothing>, <nothing>]]"; expected "Iterable[Dict[int, MyTypedDict]]"`.

But it doesn't do the same for the first one.

**Expected Behavior**

No error [OR] The same error for the first case.

**Actual Behavior**

(Write what happened.)

**Your Environment**

- Mypy version used: `0.812`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: `3.9.4`
- Operating system and version: macOS Big Sur 11.3

**Comment by svisser** [comment]
mypy doesn't report an error anymore it seems (checked with mypy 1.11.0 and Python 3.11.0).
